File: Week1_Part2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>


int main(int argc, string argv[])
{
    if (argc != 2)
    {
    return 1;
    }

    int key = atoi(argv[1]);
    string plaintext = get_string("Give me a string to cipher: ");
    char ciphertext[strlen(plaintext)];

    for (int i = 0; i < strlen(plaintext); i++)
    {
        if (isupper(plaintext[i]))
        {
            ciphertext[i] = ((plaintext[i] + key) - 65) % 26 + 65;
        }

        else if (islower(plaintext[i]))
        {
            ciphertext[i] = ((plaintext[i] + key) - 97) % 26 + 97;
        }
        else
        {
            ciphertext[i] = plaintext[i];
        }
    }
    printf("%s: " "%s\n", "plaintext", plaintext);
    printf("%s: " "%s\n", "ciphertext", ciphertext);
}

File: Week5/Sentimental/caesar/caesar.py
import sys


ALPHABET_LENGTH = 26
ASCII_UPPERCASE_A = 65
ASCII_LOWERCASE_A = 97


def encryptWord(word, key):
    encryptedWord = []

    for char in word:
        if not char.isalpha():
            encryptedWord.append(char)
        elif char.isupper():
            encryptedLetter = getEncryptedLetter(ASCII_UPPERCASE_A, char, key)
            encryptedWord.append(encryptedLetter)
        else:
            encryptedLetter = getEncryptedLetter(ASCII_LOWERCASE_A, char, key)
            encryptedWord.append(encryptedLetter)

    return ''.join(encryptedWord)


def getEncryptedLetter(asciiCaseA, char, key):
    letter = ((ord(char) + key) - asciiCaseA) % ALPHABET_LENGTH + asciiCaseA
    return str(chr(letter))


def main():
    if len(sys.argv) != 2:
        exit(1)

    key = sys.argv[1]
    plaintext = input("Give me a word to cipher!: ")
    ciphertext = encryptWord(plaintext, int(key))
    print("plaintext: ", plaintext)
    print("ciphertext: ", ciphertext)


if __name__ == '__main__':
    main()
